File: LanPlatform/platformJS/Content/ContentItem.js
LPContent.ContentType = {
    None: 0,
    Binary: 1,
    ImageBmp: 2,
    ImageGif: 3,
    ImageJpg: 4,
    ImagePng: 5,
    ImageSvg: 6
}

LPContent.ContentItem = function ()
{
    this.Id = 0;
    this.Version = 0;
    this.Author = 0;
    this.Hash = "";
    this.Filename = "";
    this.Size = 0;
    this.Type = LPContent.ContentType.None;
    this.TimeAdded = 0;
}

LPContent.ContentItem.prototype.LoadModel = function (model) {
    if (this.Id != 0) {
        return;
    }

    this.Id = model.Id;
    this.Version = model.Version;
    this.Author = model.Author;
    this.Hash = model.Hash;
    this.Filename = model.Filename;
    this.Size = model.Size;
    this.Type = model.Type;
    this.TimeAdded = model.TimeAdded;

    return;
}

File: LanPlatform/Content/ContentItem.cs
using System;
using GabionPlatform.Database;

namespace GabionPlatform.Content
{
    public class ContentItem : EditableDatabaseObject
    {
        public long Owner { get; set; }
        public bool Visible { get; set; }
        public String Hash { get; set; }
        public String Filename { get; set; }
        public long Size { get; set; }
        public ContentType Type { get; set; }
        public long TimeAdded { get; set; }

        public String DataMime
        {
            get
            {
                String mime = "";

                switch (Type)
                {
                    case ContentType.Binary:
                    {
                        mime = "application/octet-stream";

                        break;
                    }

                    case ContentType.ImageBmp:
                    {
                        mime = "image/bmp";

                        break;
                    }

                    case ContentType.ImageGif:
                    {
                        mime = "image/gif";

                        break;
                    }

                    case ContentType.ImageJpg:
                    {
                        mime = "image/jpeg";

                        break;
                    }

                    case ContentType.ImagePng:
                    {
                        mime = "image/png";

                        break;
                    }
                    
                    case ContentType.ImageSvg:
                    {
                        mime = "image/svg+xml";

                        break;
                    }
                }

                return mime;
            }
        }

        public ContentItem()
        {
            Owner = 0;
            Visible = true;
            Hash = "";
            Filename = "";
            Size = 0;
            Type = ContentType.None;
            TimeAdded = 0;
        }

        public bool IsImage()
        {
            bool success = false;

            switch (Type)
            {
                case ContentType.ImageBmp:
                case ContentType.ImageGif:
                case ContentType.ImageJpg:
                case ContentType.ImagePng:
                case ContentType.ImageSvg:
                {
                    success = true;

                    break;
                }
            }

            return success;
        }
    }

    public enum ContentType
    {
        None = 0,
        Binary,
        ImageBmp,
        ImageGif,
        ImageJpg,
        ImagePng,
        ImageSvg
    }
}